api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java

```diff
@@ -427,23 +423,4 @@ protected void validateTerm(String term) {
     // term has 2 words AND has [*word and -word]
     // term has 2 or more words AND has [*word and *word2]
   }
-
-  protected AgeType validateAgeType(String age) {
-    return Optional.ofNullable(age)
-        .map(AgeType::fromValue)
-        .orElseThrow(
-            () ->
-                new BadRequestException(
-                    String.format(BAD_REQUEST_MESSAGE, "age type parameter", age)));
-  }
-
-  protected GenderOrSexType validateGenderOrSexType(String genderOrSex) {
-    return Optional.ofNullable(genderOrSex)
-        .map(GenderOrSexType::fromValue)
-        .orElseThrow(
-            () ->
-                new BadRequestException(
-                    String.format(
-                        BAD_REQUEST_MESSAGE, "gender or sex at birth parameter", genderOrSex)));
-  }
 }
```

Comment:
I think these validate methods should stay with the CohortBuilderController. It doesn't make sense to move them into the service layer since you removed them from the ChartServiceImpl#findCohortReviewDemoChartInfo(Set<Long> participantIds). I think it's also good practice to throw BadRequestExceptions from the controller and fail fast instead of failing in layers below.

Reply:
done